src/linker/Linker.Dataflow/FlowAnnotations.cs

```diff
@@ -28,12 +28,16 @@ public FlowAnnotations (LinkContext context)
 		}
 
 		public bool RequiresDataFlowAnalysis (MethodDefinition method) =>
+			GetAnnotations (method.DeclaringType).TryGetAnnotation (method, out var methodAnnotations)
+				&& (methodAnnotations.ReturnParameterAnnotation != DynamicallyAccessedMemberTypes.None || methodAnnotations.ParameterAnnotations != null);
```

Comment:
This will have to work differently in NativeAOT - where we might have to rely on dataflow to handle generic argument dataflow - it's not clear yet though.